From 9894667b99602088143c59f9aba44d6d780a091f Mon Sep 17 00:00:00 2001
From: Igor Grinberg <igrinberg@habana.ai>
Date: Thu, 28 Jan 2021 17:38:00 +0200
Subject: [PATCH] [SW-0] gaudi: security: use common nics defs

Move security code to use common NICs definitions.
This in preparation for aligning the reset code which will also use the
same NICs definitions.

Change-Id: Ia95ebf0a3928031abb35fd0061ac9082bd8a1293
Signed-off-by: Igor Grinberg <igrinberg@habana.ai>
---

diff --git a/gaudi/common.h b/gaudi/common.h
index 3869a9b..5fede8f 100644
--- a/gaudi/common.h
+++ b/gaudi/common.h
@@ -33,7 +33,9 @@
 #define I2C_SLAVE_DEFAULT_INTR_MASK	0x8FF
 
 #define GAUDI_NICS_NUM			5
+#define GAUDI_NICS_DIST			0x40000
 #define GAUDI_NIC_ENGS_NUM		(GAUDI_NICS_NUM * 2)
+#define GAUDI_NIC_ENGS_DIST		0x1000
 #define GAUDI_QPCS_NUM			2
 #define GAUDI_TPCS_NUM			8
 #define GAUDI_MMES_NUM			4
diff --git a/gaudi/security.c b/gaudi/security.c
index f570e4c..571f905 100644
--- a/gaudi/security.c
+++ b/gaudi/security.c
@@ -60,6 +60,8 @@
 #include <spi.h>
 #include <sha256.h>
 
+#include "common.h"
+
 #define PROT_REGS_COUNT		32
 #define PROT_REGS_MAX		(PROT_REGS_COUNT - 1) /* Secure config -> LKD */
 #define BITS_IN_REG		32
@@ -867,18 +869,16 @@
 	}
 }
 
-#define NICS_NUM	5
-#define NICS_DIST	0x40000
-#define NIC_ENGS_DIST	0x1000
 static void preboot_secure_set_nic_eng_role(u32 prot_reg, u32 mask, int eng_cnt)
 {
 	int i, j;
 	u32 reg;
 
-	for (i = 0; i < NICS_NUM; i++) {
+	for (i = 0; i < GAUDI_NICS_NUM; i++) {
 		for (j = 0; j < eng_cnt; j++) {
 			/* Set the engine role (security level) */
-			reg = prot_reg + NICS_DIST * i + NIC_ENGS_DIST * j;
+			reg = prot_reg + GAUDI_NICS_DIST * i +
+							GAUDI_NIC_ENGS_DIST * j;
 			regmap_writel(mask, reg);
 		}
 	}
@@ -900,9 +900,10 @@
 	int i, j;
 	u32 reg;
 
-	for (i = 0; i < NICS_NUM; i++) {
+	for (i = 0; i < GAUDI_NICS_NUM; i++) {
 		for (j = 0; j < eng_cnt; j++) {
-			reg = base_reg + NICS_DIST * i + NIC_ENGS_DIST * j;
+			reg = base_reg + GAUDI_NICS_DIST * i +
+							GAUDI_NIC_ENGS_DIST * j;
 			/* Protect the engine configuration */
 			regmap_writel(mask, PRIV_PROT_REG(reg));
 			/* Protect the private prot bits and config space */
@@ -927,8 +928,8 @@
 	u32 nic_off;
 	u64 base;
 
-	for (i = 0; i < NICS_NUM; i++) {
-		nic_off = NICS_DIST * i;
+	for (i = 0; i < GAUDI_NICS_NUM; i++) {
+		nic_off = GAUDI_NICS_DIST * i;
 
 		/* STAT */
 		base = nic_off + mmNIC0_STAT_BASE;
